integration/testnet/network.go

```diff
@@ -533,19 +544,34 @@ func (net *FlowNetwork) AddNode(t *testing.T, bootstrapDir string, nodeConf Cont
 			// uncomment line below to point the access node exclusively to a single collection node
 			// nodeContainer.addFlag("static-collection-ingress-addr", "collection_1:9000")
 			nodeContainer.addFlag("collection-ingress-port", "9000")
-			nodeContainer.opts.HealthCheck = testingdock.HealthCheckCustom(healthcheckAccessGRPC(hostGRPCPort))
-			nodeContainer.Ports[AccessNodeAPIPort] = hostGRPCPort
-			nodeContainer.Ports[AccessNodeAPIProxyPort] = hostHTTPProxyPort
-			net.AccessPorts[AccessNodeAPIPort] = hostGRPCPort
-			net.AccessPorts[AccessNodeAPIProxyPort] = hostHTTPProxyPort
+
+			var accessNodeAPIPortKey, accessNodeAPIProxyPortKey string
+			if nodeConf.Unstaked {
+				accessNodeAPIPortKey = UnstakedAccessNodeAPIPort
+				accessNodeAPIProxyPortKey = UnstakedAccessNodeAPIProxyPort
+			} else {
+				nodeContainer.opts.HealthCheck = testingdock.HealthCheckCustom(healthcheckAccessGRPC(hostGRPCPort))
+				accessNodeAPIPortKey = AccessNodeAPIPort
+				accessNodeAPIProxyPortKey = AccessNodeAPIProxyPort
+			}
+
+			nodeContainer.Ports[accessNodeAPIPortKey] = hostGRPCPort
+			nodeContainer.Ports[accessNodeAPIProxyPortKey] = hostHTTPProxyPort
+			net.AccessPorts[accessNodeAPIPortKey] = hostGRPCPort
+			net.AccessPorts[accessNodeAPIProxyPortKey] = hostHTTPProxyPort
```

Comment:
I still have confusions about this code.

If my test network has multiple nodes of the same time (e.g consensus), then each time we reach this block of code we've generated a new port number, so we would overwrite the old one in the `net.AccessPorts` map right? In other words, `net.AccessPorts` only contains the port number for the instance of the node type which was created last?